Assert failed: No more than 1024 pending puts are allowed on a single channel.

I have a lot of events in store and some client, which tries to read them from the very beginning.
This results in:
```
.....
ring.sse/start-dispatch-loop/fn/state-machine--auto--/fn      sse.clj:   75
ring.sse/send-event                                           sse.clj:   64
clojure.core.async/put!                                     async.clj:  165
clojure.core.async.impl.channels.ManyToManyChannel/put!  channels.clj:  152
java.lang.AssertionError: Assert failed: No more than 1024 pending puts are allowed on a single channel. Consider using a windowed buffer.
                          (< (.size puts) impl/MAX-QUEUE-SIZE)
```

The default `put-fn` in `send-envent` is actually `async/>!!`, but for some reason `start-dispatch-loop` explicitly uses `async/put!`.

We need to be able to specify `async/>!` as `put-fn`.

Besides that, I think `async/>!` would be better default behavior, because it allows backpressure.

So why is it `async/put!` is used? And what would be better way to allow `async/>!`?



Edit: `async/>!` of course - we are in `go` block!

--- a/src/ring/sse.clj
+++ b/src/ring/sse.clj
@@ -88,7 +88,7 @@
                 (if (map? event)
                   (reduce (fn [agg [k v]] (assoc agg k (str v))) {} event)
                   {:data (str event)})]
-            (when (send-event response-channel event-name event-data event-id async/put! raise)
+            (when (send-event response-channel event-name event-data event-id raise)
               (recur))))))
     (async/close! event-channel)
     (async/close! response-channel)

--- a/src/ring/sse.clj
+++ b/src/ring/sse.clj
@@ -88,7 +88,7 @@
                 (if (map? event)
                   (reduce (fn [agg [k v]] (assoc agg k (str v))) {} event)
                   {:data (str event)})]
-            (when (send-event response-channel event-name event-data event-id raise)
+            (when (send-event response-channel event-name event-data event-id async/>! raise)
               (recur))))))
     (async/close! event-channel)
     (async/close! response-channel)

